### DISCUSSION TITLE
PiGpio DigitalOutputProvider context building fails. Please help...

### DISCUSSION THREAD

**Discussion by crogialli** [Pi4J/pi4j#357] [Q&A] [answered]
My code:

55		PiGpio piGpio = PiGpio.newNativeInstance();
56		pi4j = Pi4J.newContextBuilder()
57			.noAutoDetect()
58			.add(PiGpioSpiProvider.newInstance(piGpio))
59			.add(PiGpioI2CProvider.newInstance(piGpio))
60			.add(PiGpioDigitalOutputProvider.newInstance(piGpio))
61			.build();

When executing, it fails while trying to add  PiGpioDigitalOutputProvider (Spi and I2c go flawlessly), because it doesn't find a field named "id" (**lowercase**):

2024-05-13 23:20:09,317 INFO  [main] Pi4J: New context builder
**Exception in thread "main" java.lang.NoSuchFieldError: id
	at com.pi4j.plugin.raspberrypi.provider.gpio.digital.RpiDigitalOutputProviderImpl.<init>(RpiDigitalOutputProviderImpl.java:47)**
	at com.pi4j.plugin.raspberrypi.provider.gpio.digital.RpiDigitalOutputProvider.newInstance(RpiDigitalOutputProvider.java:50)
	at org.codehaus.groovy.vmplugin.v8.IndyInterface.fromCache(IndyInterface.java:318)
	at kabasys.hardware.RaspberryPi4JV2HardManager.<init>(RaspberryPi4JV2HardManager.groovy:60)

I tried to use RpiDigitalOutputProvider instead of PiGpioDigitalOutputProvider --> same error
I tried to use Pi4J.newAutoContext --> same error

What am I doing wrong?
Thank you in advance for your support.

**Comment by crogialli** [comment, marked as answer]
Ok, my fault.
I'm adding Pi4J V2 support to an app that can also run Pi4J V1 as legacy mode.
An exceeding V1 include in the V2 class was causing the issue.

Sorry guys.

**Reply by FDelporte** [reply]
No sorry, it's always nice when someone is able to solve a problem by themselves and report it here! Have fun with Pi4J!

**Comment by taartspi** [comment]
This AM I was going to reread this post and view your question and I saw you already resolved your problem.    I don't know how you are adding V2 support, but I will mention the webpage https://pi4j.com/documentation/providers/ explains the available automatic loading of Providers as implemented in pi4j 2.5.0-SNAPSHOT/
